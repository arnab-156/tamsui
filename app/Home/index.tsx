import React from 'react';

export default function Home(): React.ReactElement {
  return (
    <h1>Hello, React SSR!</h1>
  );
}
