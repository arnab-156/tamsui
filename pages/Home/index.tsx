import React from 'react';
import { Link } from 'react-router-dom';

import Layout from 'pages/Layout';

import styles from './styles.module.scss';

export default function Home(): React.ReactElement {
  return (
    <Layout>
      <img className={styles.logo} alt="Tamsui logo" src="/static/images/logo.webp" />
      <h1 className={styles.heading}>Tamsui</h1>
      <Link to="/counter">React Counter</Link>
    </Layout>
  );
}
