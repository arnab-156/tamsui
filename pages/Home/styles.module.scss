.logo {
  display: block;
  margin: 0 auto;
  width: 200px;
  height: auto;
}

.heading {
  text-align: center;  
}
