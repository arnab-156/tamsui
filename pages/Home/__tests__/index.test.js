import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import Home from '../index';

function RouterWrappedHome() {
  return (
    <MemoryRouter>
      <Home />
    </MemoryRouter>
  );
}

describe('Home Component', () => {
  test('matches snapshot', () => {
    const tree = renderer.create(<RouterWrappedHome />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('contains the hello text', () => {
    render(<RouterWrappedHome />);

    expect(screen.getByRole('heading')).toHaveTextContent('Hello, React SSR!');
  });
});
