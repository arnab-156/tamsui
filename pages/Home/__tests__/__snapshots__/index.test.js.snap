// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component matches snapshot 1`] = `
<main>
  <h1>
    Hello, React SSR!
  </h1>
  <a
    href="/counter"
    onClick={[Function]}
  >
    React Counter
  </a>
</main>
`;
